### PR TITLE
CT-1456 Configure State Machine for Non Offender Non Trigger …

### PR DESCRIPTION
**Pull request by stephenrichards**
…SAR cases

### CODE REVIEW

File: app/services/current_team_and_user/foi.rb [hunk 1]
```diff
@@ -1,55 +1,4 @@
...
-    def responded
-      @team = @case.managing_team
-      @user = nil
-    end
-
-    def closed
-      @team = nil
-      @user = nil
-    end
+  class FOI < Base
```
Comment (misaka) [inline app/services/current_team_and_user/foi.rb]: Use the full path/name of `Base` -> `CurrentTeamAndUser::Base`. Ruby doesn't fix the issue of trying to resolve class names in the top namespace until 2.5, which I think could cause a side-effect here.

File: app/services/current_team_and_user/sar.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module CurrentTeamAndUser
+  class SAR < Base
```
Comment (misaka) [inline app/services/current_team_and_user/sar.rb]: Use the full path/name of `Base` -> `CurrentTeamAndUser::Base`. Ruby doesn't fix the issue of trying to resolve class names in the top namespace until 2.5, which I think could cause a side-effect here.

File: app/services/current_team_and_user/foi.rb (hunk 1 above)
Comment (misaka) [reply app/services/current_team_and_user/foi.rb]: https://blog.bigbinary.com/2017/10/18/ruby-2.5-has-removed-top-level-constant-lookup.html

**Review by misaka — CHANGES_REQUESTED** [review]
Need to go afk, will continue reviewing shortly.

File: lib/tasks/sar.rake [hunk 3]
```diff
@@ -0,0 +1,42 @@
+namespace :sar do
```
Comment (misaka) [inline lib/tasks/sar.rake]: We use the `cts` script to generate FOIs, we should put this in there.

File: spec/state_machines/foi_permitted_events_spec.rb [hunk 4]
```diff
@@ -0,0 +1,193 @@
+require 'rails_helper'
+
+describe 'FOI permittted events' do
```
Comment (misaka) [inline spec/state_machines/foi_permitted_events_spec.rb]: This should specify what method is being tested: `ConfigurableStateMachine::permitted_events` (I think?)

File: spec/state_machines/sar_permitted_events_spec.rb [hunk 5]
```diff
@@ -0,0 +1,195 @@
+require 'rails_helper'
+
+describe 'SAR permitted events' do
```
Comment (misaka) [inline spec/state_machines/sar_permitted_events_spec.rb]: This should specify what method is being is being tested `ConfigurableStateMachine::permitted_events`.

**Review by misaka — CHANGES_REQUESTED** [review]

File: lib/tasks/sar.rake (hunk 3 above)
Comment (stephenrichards) [reply lib/tasks/sar.rake]: This is a temprorary script which I can use until that's done.

**Review by misaka — APPROVED** [review]

**Pull request merged**